Skip setting read-only parameters during model export

The exporter tries to write read-only variables directly.
 * To reproduce: try exporting the `sldemo_clutch_import.simulink` model.
 * Stacktrace: 
```
!ENTRY org.eclipse.core.jobs 4 2 2018-07-26 09:07:59.376
!MESSAGE An internal error occurred during: "Exporting model sldemo_clutch_import...".
!STACK 0
hu.bme.mit.massif.communication.CommandEvaluationException: Exception occurred while evaluating command!
	at hu.bme.mit.massif.communication.AbstractCommandEvaluator.evaluateCommand(AbstractCommandEvaluator.java:65)
	at hu.bme.mit.massif.communication.AbstractCommandEvaluator.evaluateCommands(AbstractCommandEvaluator.java:264)
	at hu.bme.mit.massif.communication.command.MatlabCommand.execute(MatlabCommand.java:118)
	at hu.bme.mit.massif.simulink.api.Exporter.exportBlocks(Exporter.java:499)
	at hu.bme.mit.massif.simulink.api.Exporter.export(Exporter.java:278)
	at hu.bme.mit.massif.simulink.ui.handlers.ExportModelHandler$1.run(ExportModelHandler.java:142)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:60)
Caused by: com.mathworks.engine.MatlabExecutionException: set_param('sldemo_clutch_import/Friction Model','IsSubsystemVirtual','on');
	at com.mathworks.engine.FutureResult.get(FutureResult.java:64)
	at com.mathworks.engine.MatlabEngine.eval(MatlabEngine.java:349)
	at hu.bme.mit.massif.communication.matlabengine.MatlabEngineAccess.executeEval(MatlabEngineAccess.java:105)
	at hu.bme.mit.massif.communication.AbstractCommandEvaluator.evaluateCommand(AbstractCommandEvaluator.java:60)
	... 6 more
Caused by: com.mathworks.mvm.exec.MvmRuntimeException: SubSystem block parameter 'IsSubsystemVirtual' is read-only
	... 10 more
```

--- a/plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/Exporter.java
+++ b/plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/Exporter.java
@@ -494,8 +494,18 @@ public class Exporter {
 
             // Set block dialog/mask parameters
             EList<Parameter> parameters = block.getParameters();
+            boolean isMaskOn = false;
             for (Parameter parameter : parameters) {
-                if(!parameter.isReadOnly()) {
+                if(parameter.getName().equals("Mask")) {
+                    isMaskOn = parameter.getValue().equals("on");
+                }
+            }
+            for (Parameter parameter : parameters) {
+                // Set if
+                // * not read-only AND
+                // * it is _not_ a mask parameter OR mask is on
+                // Explanation: if the mask is off, mask parameters should be ignored
+                if(!parameter.isReadOnly() && (isMaskOn || !parameter.getName().startsWith("Mask"))) {
                     String propertyName = parameter.getName();
                     commandFactory.setParam().addParam(getFQN(block)).addParam(propertyName).addParam(parameter.getValue()).execute();                    
                 }

--- a/plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/Exporter.java
+++ b/plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/Exporter.java
@@ -495,8 +495,10 @@ public class Exporter {
             // Set block dialog/mask parameters
             EList<Parameter> parameters = block.getParameters();
             for (Parameter parameter : parameters) {
-                String propertyName = parameter.getName();
-                commandFactory.setParam().addParam(getFQN(block)).addParam(propertyName).addParam(parameter.getValue()).execute();
+                if(!parameter.isReadOnly()) {
+                    String propertyName = parameter.getName();
+                    commandFactory.setParam().addParam(getFQN(block)).addParam(propertyName).addParam(parameter.getValue()).execute();                    
+                }
             }
 
         }

--- a/plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/adapter/block/DefaultBlockAdapter.java
+++ b/plugins/hu.bme.mit.massif.simulink.api/src/hu/bme/mit/massif/simulink/api/adapter/block/DefaultBlockAdapter.java
@@ -72,6 +72,10 @@ public class DefaultBlockAdapter implements IBlockAdapter {
 
         	IVisitableMatlabData value = entry.getValue();
 			Parameter prop = SimulinkFactory.eINSTANCE.createParameter();
+			if (propertyName.matches(".*READONLY$")) {
+                propertyName = propertyName.replace("_READONLY", "");
+                prop.setReadOnly(true);
+            }
 			prop.setName(propertyName);
 			
 			if (value == null) {

